src/main/scala/mesosphere/marathon/api/v2/AppNormalization.scala

```diff
@@ -297,7 +297,22 @@ object AppNormalization {
     )
   }
 
-  def apply(config: Config): Normalization[App] = Normalization { app =>
+  def forPreValidation(config: Config): Normalization[App] = Normalization{ app =>
```

Comment:
We need a comment here about why we had to split normalization in two. Something like:
```suggestion
/** 
 * This is a partial normalization, which currently only normalizes the [[App.role]] and [[App.acceptedResourceRoles]] fields.
 * We do this because app validation is relying on these fields being set correctly.
 */
  def forPreValidation(config: Config): Normalization[App] = Normalization{ app =>
```